spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/config/s3/S3ConfigDataLocationResolver.java

```diff
@@ -0,0 +1,131 @@
+
+	/**
+	 * AWS S3 Config Data prefix.
+	 */
+	public static final String PREFIX = "aws-s3:";
+
+	private final Log log;
+
+	public S3ConfigDataLocationResolver(DeferredLogFactory deferredLogFactory) {
+		this.log = deferredLogFactory.getLog(S3ConfigDataLocationResolver.class);
+	}
+
+	@Override
+	protected String getPrefix() {
+		return PREFIX;
+	}
+
+	@Override
+	public List<S3ConfigDataResource> resolveProfileSpecific(ConfigDataLocationResolverContext resolverContext,
+			ConfigDataLocation location, Profiles profiles) throws ConfigDataLocationNotFoundException {
+
+		S3Properties s3Properties = loadProperties(resolverContext.getBinder());
+
+		registerBean(resolverContext, AwsProperties.class, loadAwsProperties(resolverContext.getBinder()));
+		registerBean(resolverContext, S3Properties.class, s3Properties);
+		registerBean(resolverContext, CredentialsProperties.class,
+				loadCredentialsProperties(resolverContext.getBinder()));
+		registerBean(resolverContext, RegionProperties.class, loadRegionProperties(resolverContext.getBinder()));
+
+		registerAndPromoteBean(resolverContext, S3Client.class, BootstrapRegistry.InstanceSupplier.of(
+				S3ClientFactory.s3Client(createS3ClientBuilder(s3Properties, resolverContext.getBootstrapContext()))));
+
+		S3PropertySources propertySources = new S3PropertySources();
+
+		List<String> contexts = getCustomContexts(location.getNonPrefixedValue(PREFIX));
+
+		List<S3ConfigDataResource> locations = new ArrayList<>();
+		contexts.forEach(propertySourceContext -> locations.add(new S3ConfigDataResource(propertySourceContext,
+				location.isOptional(), s3Properties.isEnableImport(), propertySources)));
+
+		if (!location.isOptional() && locations.isEmpty()) {
+			throw new S3KeysMissingException("No S3 keys provided in `spring.config.import=aws-s3:` configuration.");
+		}
+
+		return locations;
+	}
+
+	private S3ClientBuilder createS3ClientBuilder(S3Properties s3Properties, BootstrapContext context) {
+		S3ClientBuilder builder = S3Client.builder();
+		Optional.ofNullable(s3Properties.getCrossRegionEnabled()).ifPresent(builder::crossRegionAccessEnabled);
+		builder.serviceConfiguration(s3Properties.toS3Configuration());
+
+		builder = configure(builder, s3Properties, context);
+
+		try {
+			AwsS3ClientCustomizer configurer = context.get(AwsS3ClientCustomizer.class);
+			if (configurer != null) {
+				AwsClientCustomizer.apply(configurer, builder);
+			}
+		}
+		catch (IllegalStateException e) {
+			log.debug("Bean of type AwsClientConfigurerParameterStore is not registered: " + e.getMessage());
+		}
+		return builder;
+	}
+
+	protected S3Properties loadProperties(Binder binder) {
+		return binder.bind(S3Properties.PREFIX, Bindable.of(S3Properties.class)).orElseGet(S3Properties::new);
+	}
+
+	static class S3ClientFactory {
+
+		static S3Client s3Client(S3ClientBuilder builder) {
+			if (ClassUtils.isPresent("io.awspring.cloud.s3.crossregion.CrossRegionS3Client", null)) {
```

Comment:
CrossRegionS3Client is deprecated, lets not use it anymore.

Reply:
This has been addressed